multi-agent/lib/mas/elements/providers/mcp_server_client/validator.py

```
@@ -91,6 +91,44 @@ async def _check_connection(
+                # real issuer, not the MCP URL. Rediscover the issuer the same
+                # way the sign-in status widget (auth.discovery) does before
+                # concluding the resource is unauthenticated.
+                if auth_cred is None and is_sign_in:
+                    try:
+                        detection = await context.auth_service.discover(str(config.mcp_url))
+                    except (CancelledError, TimeoutError):
+                        # Real budget exhaustion — let it propagate so the
+                        # outer handler reports it as a timeout, not a
+                        # misleading "connection failed".
+                        raise
+                    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)